Factorio-Biter-Battles/Factorio-Biter-Battles
# Implement task system for captain game

### Brief description of the changes:
Create tasks list in captain event.
The first player joining the task group can edit the name, other players can't.
The captain can edit all task group name and bypass this security above.

Example of view : 
Button added on top : 
![image](https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/assets/95619848/3c5122a2-132a-47c7-a520-b830051fabae)
If you join a task  : 
![image](https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/assets/95619848/bbf37a86-d444-45a6-a2c9-97185ebdcdf0)

If you have no task : 
![image](https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/assets/95619848/f1262537-2741-4829-aab2-24f26c1d66d5)

View of captain : 
![image](https://github.com/Factorio-Biter-Battles/Factorio-Biter-Battles/assets/95619848/fd1bd112-368f-42e5-a88b-eb386f4731a1)

### To test it : 
Create a local captain game event, then use manager to switch team, to see the GUI. 
You can do that to see the captain view of it (just close gui and open it again to have correct view ) : **/c global.special_games_variables["captain_mode"]["captainList"][1]="yourPlayerName"**

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.
